adal/authentication_context.py

```diff
@@ -75,11 +75,17 @@ def __init__(
             read timeout) <timeouts>` tuple.
         :param enable_pii: (optional) Unless this is set to True,
             there will be no Personally Identifiable Information (PII) written in log.
+        :param verify_ssl: (optional) requests verify. Either a boolean, in which case it 
+            controls whether we verify the server's TLS certificate, or a string, in which 
+            case it must be a path to a CA bundle to use. This value is ignored if env 
+            variable ADAL_PYTHON_SSL_NO_VERIFY is used.

```

Comment:
Would it be more natural that we do the other way around, i.e. env var will be ignored if function argument is explicitly provided? I know there might not be an industry standard on this behavior but, it happens that the [Python standard lib `urllib`](https://docs.python.org/2/library/urllib.html) uses the same approach as what I suggested here:

> Alternatively, the optional proxies argument may be used to explicitly specify proxies. It must be a dictionary mapping scheme names to proxy URLs, where an empty dictionary causes no proxies to be used, and None (the default value) causes environmental proxy settings to be used as discussed above. For example: ...

Reply:
I agree with @rayluo on this

Reply:
Ok, in progress